src/main.cpp

```diff
@@ -33,63 +37,136 @@ constexpr auto VERSION = _VERSION;
 #else
 constexpr auto VERSION = "[Unkown version]";
 #endif
+class InputParser {
```

Comment:
I think that this class could be moved to its own file in `utils/args-parser.h/cpp` so `main.cpp` looks simpler.

Reply:
Will do.

Reply:
I'll also rename the class.

Reply:
Let me know if you need help. I can do this one :D

Reply:
Not a problem, WIP right now.